src/axolotl/monkeypatch/llama_attn_hijack_flash.py

```diff
@@ -194,7 +194,7 @@ def flashattn_forward(
         # only on first autoregressive step q,k,v have same seqlen
         is_causal = key_states.shape == query_states.shape
 
-    if cu_seqlens is not None and max_seqlen is not None:
+    if cu_seqlens is not None and max_seqlen is not None and cu_seqlens.dim() == 1:
```

Comment:
Does this not work without this change? I don't think any of your code uses sample packing, so I'd prefer not to break other things and we can address if this change is correct in another PR

Reply:
This was required for `sample_packing: true` to work for me.

`sample_packing: false` appeared to work without this change.

I have little understanding of why for either case.

Reply:
Without it:

`accelerate launch ./scripts/finetune.py ./examples/llama-2/lora.yml`:

```
[2023-09-12 04:25:44,129] [INFO] [axolotl.utils.dataloader.generate_batches:181] [PID:3315] generating packed batches                                       | 0/1 [00:00<?, ?it/s]
[2023-09-12 04:25:44,129] [INFO] [axolotl.utils.dataloader.generate_batches:187] [PID:3315] 0309058c71626f624864fe0a6b1d7bbb52a0712fb39841fc02e499e97eefc0f4
[2023-09-12 04:25:44,130] [INFO] [axolotl.utils.dataloader._len_est:262] [PID:3315] packing_efficiency_estimate: 0.98 total_num_tokens per device: 4899
A decoder-only architecture is being used, but right-padding was detected! For correct generation results, please set `padding_side='left'` when initializing the tokenizer.
  0%|                                                                                                                                                      | 0/1 [00:01<?, ?it/s]
Traceback (most recent call last):
  File "/workspace/axolotl/./scripts/finetune.py", line 283, in <module>
    fire.Fire(do_cli)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/workspace/axolotl/./scripts/finetune.py", line 279, in do_cli
    train(cfg=parsed_cfg, cli_args=parsed_cli_args, dataset_meta=dataset_meta)
  File "/workspace/axolotl/src/axolotl/train.py", line 118, in train
    trainer.train(resume_from_checkpoint=resume_from_checkpoint)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer.py", line 1574, in train
    return inner_training_loop(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer.py", line 1966, in _inner_training_loop
    self._maybe_log_save_evaluate(tr_loss, model, trial, epoch, ignore_keys_for_eval)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer.py", line 2299, in _maybe_log_save_evaluate
    metrics = self.evaluate(ignore_keys=ignore_keys_for_eval)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer.py", line 3053, in evaluate
    self.control = self.callback_handler.on_evaluate(self.args, self.state, self.control, output.metrics)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer_callback.py", line 388, in on_evaluate
    return self.call_event("on_evaluate", args, state, control, metrics=metrics)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/trainer_callback.py", line 406, in call_event
    result = getattr(callback, event)(
  File "/workspace/axolotl/src/axolotl/utils/callbacks.py", line 485, in on_evaluate
    log_table_from_dataloader("Eval", eval_dataloader)
  File "/workspace/axolotl/src/axolotl/utils/callbacks.py", line 454, in log_table_from_dataloader
    predictions = trainer.model.generate(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/peft/peft_model.py", line 988, in generate
    outputs = self.base_model.generate(**kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/generation/utils.py", line 1602, in generate
    return self.greedy_search(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/generation/utils.py", line 2450, in greedy_search
    outputs = self(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/accelerate/hooks.py", line 165, in new_forward
    output = old_forward(*args, **kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 820, in forward
    outputs = self.model(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/accelerate/hooks.py", line 165, in new_forward
    output = old_forward(*args, **kwargs)
  File "/workspace/axolotl/src/axolotl/monkeypatch/llama_attn_hijack_flash.py", line 531, in llama_model_forward
    layer_outputs = decoder_layer(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/accelerate/hooks.py", line 165, in new_forward
    output = old_forward(*args, **kwargs)
  File "/workspace/axolotl/src/axolotl/monkeypatch/llama_attn_hijack_flash.py", line 609, in forward
    hidden_states, self_attn_weights, present_key_value = self.self_attn(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/accelerate/hooks.py", line 165, in new_forward
    output = old_forward(*args, **kwargs)
  File "/workspace/axolotl/src/axolotl/monkeypatch/llama_attn_hijack_flash.py", line 206, in flashattn_forward
    output = flash_attn_varlen_qkvpacked_func(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/flash_attn/flash_attn_interface.py", line 648, in flash_attn_varlen_qkvpacked_func
    return FlashAttnVarlenQKVPackedFunc.apply(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/torch/autograd/function.py", line 506, in apply
    return super().apply(*args, **kwargs)  # type: ignore[misc]
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/flash_attn/flash_attn_interface.py", line 211, in forward
    out, q, k, v, out_padded, softmax_lse, S_dmask, rng_state = _flash_attn_varlen_forward(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/flash_attn/flash_attn_interface.py", line 68, in _flash_attn_varlen_forward
    out, q, k, v, out_padded, softmax_lse, S_dmask, rng_state = flash_attn_cuda.varlen_fwd(
TypeError: varlen_fwd(): incompatible function arguments. The following argument types are supported:
    1. (arg0: torch.Tensor, arg1: torch.Tensor, arg2: torch.Tensor, arg3: Optional[torch.Tensor], arg4: torch.Tensor, arg5: torch.Tensor, arg6: int, arg7: int, arg8: float, arg9: float, arg10: bool, arg11: bool, arg12: bool, arg13: Optional[torch.Generator]) -> List[torch.Tensor]

Invoked with: tensor([[[ 0.1094, -0.4062,  0.7617,  ..., -3.0625,  1.6406, -0.3047],
         [-0.1426,  0.2793, -0.6172,  ...,  0.4785, -0.6758,  0.5312],
         [-0.0269,  0.3086,  0.3770,  ...,  0.3887,  0.3770,  0.3691],
         ...,
         [ 0.6797, -1.6016, -0.4512,  ...,  1.6250,  0.5742,  1.2812],
         [ 0.7305,  1.0391, -0.6328,  ..., -0.4688,  0.5078,  0.4785],
         [-1.3594,  1.3516, -0.2891,  ..., -0.1157,  0.6484,  0.1357]],

        [[ 1.2344, -1.5547,  0.8516,  ..., -0.3828,  0.1846, -0.5547],
         [ 0.0347, -0.0273, -0.5195,  ...,  0.5977, -1.0391,  0.6797],
         [-0.1777, -0.5938, -0.3945,  ...,  0.4629,  0.4395,  0.4609],
         ...,
         [-2.4219, -2.1562, -1.8906,  ...,  1.8672,  0.8281,  2.0469],
         [-1.5469,  2.3438, -2.3906,  ..., -0.3555,  0.5820,  0.5977],
         [-0.8984, -0.4434, -1.0312,  ..., -0.2988,  0.6094, -0.4355]],

        [[ 1.9375, -1.7188, -0.7539,  ..., -0.7695,  0.5547, -1.0234],
         [ 0.0049, -0.1465, -0.2852,  ...,  0.1475, -0.5820,  0.2158],
         [-0.7031, -0.6016,  0.3047,  ...,  0.1943,  0.1846,  0.2139],
         ...,
         [-2.5469, -0.3105, -2.0469,  ...,  1.0469,  0.6797,  1.8281],
         [-2.8125,  1.1172, -2.6250,  ..., -0.3691,  0.6289,  0.6484],
         [-0.0703, -0.3164, -0.8359,  ..., -0.0674,  0.3047, -0.4062]],

        ...,

        [[ 0.7422, -0.7812,  0.7188,  ..., -0.3281,  0.0095, -0.6562],
         [-0.1641, -0.0547, -1.6797,  ..., -0.1885, -0.2637, -0.1748],
         [-0.5039, -0.7891, -0.0645,  ..., -0.0845, -0.0623, -0.0422],
         ...,
         [-1.3281, -1.3203, -1.1016,  ...,  0.5781, -0.2969,  0.3184],
         [-1.6094,  2.2812, -2.5469,  ..., -0.3145,  0.6094,  0.6289],
         [-0.9805, -0.7578, -1.3125,  ..., -0.1011,  0.0356, -0.0349]],

        [[ 0.7422, -0.7812,  0.7188,  ..., -0.3281,  0.0095, -0.6562],
         [-0.1641, -0.0547, -1.6797,  ..., -0.1885, -0.2637, -0.1748],
         [-0.5039, -0.7891, -0.0645,  ..., -0.0845, -0.0623, -0.0422],
         ...,
         [-1.3281, -1.3203, -1.1016,  ...,  0.5781, -0.2969,  0.3184],
         [-1.6094,  2.2812, -2.5469,  ..., -0.3145,  0.6094,  0.6289],
         [-0.9805, -0.7578, -1.3125,  ..., -0.1011,  0.0356, -0.0349]],

        [[ 0.7422, -0.7812,  0.7188,  ..., -0.3281,  0.0095, -0.6562],
         [-0.1641, -0.0547, -1.6797,  ..., -0.1885, -0.2637, -0.1748],
         [-0.5039, -0.7891, -0.0645,  ..., -0.0845, -0.0623, -0.0422],
         ...,
         [-1.3281, -1.3203, -1.1016,  ...,  0.5781, -0.2969,  0.3184],
         [-1.6094,  2.2812, -2.5469,  ..., -0.3145,  0.6094,  0.6289],
         [-0.9805, -0.7578, -1.3125,  ..., -0.1011,  0.0356, -0.0349]]],
       device='cuda:0', dtype=torch.bfloat16), tensor([[[-4.4141e-01, -9.2773e-03,  6.2988e-02,  ...,  5.8105e-02,
          -2.0752e-02, -9.9609e-02],
         [ 1.3203e+00,  1.0625e+00, -4.1406e-01,  ...,  4.1992e-01,
          -2.5781e-01,  4.1797e-01],
         [ 7.6294e-03, -2.6172e-01, -4.2188e-01,  ..., -4.6387e-02,
           5.6250e-01,  8.0859e-01],
         ...,
         [-1.9287e-02,  4.7607e-03,  1.7456e-02,  ...,  4.2383e-01,
           9.1406e-01, -4.5703e-01],
         [ 1.6016e-01, -5.3125e-01, -3.2031e-01,  ..., -2.7344e-01,
           1.1523e-01,  1.2061e-01],
         [-4.3555e-01,  1.2109e+00,  5.0659e-03,  ...,  6.9531e-01,
           2.6953e-01,  2.3926e-01]],

        [[-8.6719e-01,  5.5859e-01,  2.4414e-02,  ...,  1.3086e-01,
           1.7285e-01,  4.3945e-01],
         [ 5.8203e-01,  5.8984e-01, -4.2188e-01,  ..., -7.3730e-02,
           2.6367e-01, -1.1230e-01],
         [-3.7500e-01, -8.9844e-02, -2.3340e-01,  ..., -3.9648e-01,
          -4.4922e-01, -4.5508e-01],
         ...,
         [ 2.4414e-01,  2.3438e-02,  8.2520e-02,  ...,  5.3125e-01,
           6.7969e-01,  2.3340e-01],
         [-1.6797e-01,  4.0430e-01, -8.2031e-01,  ...,  1.6797e-01,
          -4.0039e-02, -4.4922e-02],
         [-2.3047e-01, -7.5781e-01, -7.7344e-01,  ...,  1.5625e+00,
          -2.6172e-01, -2.7313e-03]],

        [[ 8.5938e-02, -2.7344e-01, -3.9844e-01,  ...,  7.4219e-02,
           2.4707e-01,  1.9043e-01],
         [-9.3750e-02,  3.3984e-01,  3.4668e-02,  ..., -4.0820e-01,
           4.6875e-01, -3.9062e-01],
         [ 3.1641e-01,  5.8594e-01, -1.0449e-01,  ...,  6.5918e-02,
          -1.2085e-02, -3.1982e-02],
         ...,
         [ 1.9238e-01,  8.4473e-02,  7.3730e-02,  ..., -3.2422e-01,
           3.5645e-02, -4.6680e-01],
         [-4.8828e-02,  7.8125e-01,  3.9453e-01,  ...,  4.7266e-01,
          -3.7305e-01, -3.7305e-01],
         [ 6.1328e-01, -1.8555e-01, -6.5430e-02,  ...,  7.3438e-01,
          -1.6895e-01,  1.4160e-01]],

        ...,

        [[-7.5000e-01,  3.6523e-01, -3.9258e-01,  ..., -2.1973e-01,
           1.7871e-01, -8.5938e-02],
         [-3.3594e-01,  4.6875e-02, -1.2500e+00,  ..., -3.3984e-01,
          -1.1768e-01, -2.3633e-01],
         [-2.1875e-01,  5.5664e-02,  2.1484e-01,  ...,  1.0938e+00,
           1.3281e+00,  1.3281e+00],
         ...,
         [ 2.9785e-02,  2.4536e-02,  2.1484e-02,  ...,  4.2383e-01,
           4.0820e-01, -1.3574e-01],
         [-7.4219e-02,  6.8359e-03, -2.4902e-01,  ..., -4.9805e-01,
          -1.2665e-03, -4.5898e-02],
         [ 3.1836e-01, -1.6992e-01,  2.7344e-02,  ..., -2.2656e-01,
           7.8125e-01,  2.8906e-01]],

        [[-7.5000e-01,  3.6523e-01, -3.9258e-01,  ..., -2.1973e-01,
           1.7871e-01, -8.5938e-02],
         [-3.3594e-01,  4.6875e-02, -1.2500e+00,  ..., -3.3984e-01,
          -1.1768e-01, -2.3633e-01],
         [-2.1875e-01,  5.5664e-02,  2.1484e-01,  ...,  1.0938e+00,
           1.3281e+00,  1.3281e+00],
         ...,
         [ 2.9785e-02,  2.4536e-02,  2.1484e-02,  ...,  4.2383e-01,
           4.0820e-01, -1.3574e-01],
         [-7.4219e-02,  6.8359e-03, -2.4902e-01,  ..., -4.9805e-01,
          -1.2665e-03, -4.5898e-02],
         [ 3.1836e-01, -1.6992e-01,  2.7344e-02,  ..., -2.2656e-01,
           7.8125e-01,  2.8906e-01]],

        [[-7.5000e-01,  3.6523e-01, -3.9258e-01,  ..., -2.1973e-01,
           1.7871e-01, -8.5938e-02],
         [-3.3594e-01,  4.6875e-02, -1.2500e+00,  ..., -3.3984e-01,
          -1.1768e-01, -2.3633e-01],
         [-2.1875e-01,  5.5664e-02,  2.1484e-01,  ...,  1.0938e+00,
           1.3281e+00,  1.3281e+00],
         ...,
         [ 2.9785e-02,  2.4536e-02,  2.1484e-02,  ...,  4.2383e-01,
           4.0820e-01, -1.3574e-01],
         [-7.4219e-02,  6.8359e-03, -2.4902e-01,  ..., -4.9805e-01,
          -1.2665e-03, -4.5898e-02],
         [ 3.1836e-01, -1.6992e-01,  2.7344e-02,  ..., -2.2656e-01,
           7.8125e-01,  2.8906e-01]]], device='cuda:0', dtype=torch.bfloat16), tensor([[[-5.8899e-03, -8.3008e-03,  5.7678e-03,  ...,  6.0425e-03,
           1.3733e-02, -7.9956e-03],
         [ 5.0659e-03,  6.4392e-03, -8.1787e-03,  ..., -8.2397e-03,
           6.8970e-03, -1.9287e-02],
         [-1.4038e-03, -1.5717e-03,  1.6846e-02,  ...,  5.9891e-04,
          -1.1230e-02,  2.4567e-03],
         ...,
         [ 6.8359e-03,  3.7537e-03, -2.8442e-02,  ...,  5.1758e-02,
           1.3977e-02, -1.6235e-02],
         [-4.3335e-03,  1.5869e-02, -1.1292e-02,  ..., -9.2773e-03,
          -4.0771e-02, -2.6367e-02],
         [-5.0964e-03, -2.8381e-03, -8.5831e-04,  ..., -5.9128e-04,
           1.2207e-04,  5.1880e-04]],

        [[ 3.5400e-03,  1.2207e-02, -1.1963e-02,  ..., -6.3782e-03,
           8.6670e-03, -1.0010e-02],
         [ 3.5706e-03,  1.0071e-03,  4.3030e-03,  ...,  4.1809e-03,
           1.2054e-03, -5.4932e-03],
         [ 4.3678e-04, -7.1411e-03,  1.4160e-02,  ..., -2.6398e-03,
           3.4027e-03, -1.1353e-02],
         ...,
         [ 1.9653e-02,  2.9175e-02, -3.2715e-02,  ...,  1.7090e-02,
          -3.7109e-02,  1.5625e-02],
         [-1.4160e-02,  1.1292e-03, -3.5095e-03,  ...,  6.3477e-03,
          -1.3184e-02,  1.3672e-02],
         [-5.1880e-04,  4.8523e-03, -1.1978e-03,  ..., -6.8665e-03,
          -1.0071e-03, -8.1253e-04]],

        [[ 9.2163e-03, -5.9204e-03, -2.5635e-03,  ..., -1.2878e-02,
           3.2349e-03,  2.5177e-03],
         [ 6.8283e-04, -1.5259e-03,  6.0120e-03,  ..., -4.3640e-03,
           2.8076e-03,  5.4932e-03],
         [-1.4114e-03,  5.1270e-03,  1.6235e-02,  ..., -8.9111e-03,
          -6.4850e-04, -5.3406e-03],
         ...,
         [ 1.9653e-02,  3.4912e-02,  5.0293e-02,  ..., -3.2471e-02,
          -6.1035e-02,  7.5684e-02],
         [-3.1586e-03,  1.0315e-02,  7.8735e-03,  ..., -3.3875e-03,
          -1.5717e-03,  4.3945e-03],
         [ 7.7515e-03, -2.2278e-03,  2.4719e-03,  ..., -2.0981e-04,
          -2.0599e-03,  1.8799e-02]],

        ...,

        [[ 2.0020e-02,  1.5747e-02,  5.5695e-04,  ...,  1.7456e-02,
          -5.8899e-03,  2.6855e-03],
         [ 3.7384e-03, -6.5918e-03, -2.0752e-03,  ..., -1.7548e-03,
          -9.0942e-03, -4.8218e-03],
         [-7.9956e-03, -8.3542e-04,  9.4604e-04,  ..., -3.9062e-03,
           1.1719e-02,  8.9645e-05],
         ...,
         [-2.2125e-04, -7.3242e-03,  2.9907e-02,  ..., -3.4485e-03,
           5.5908e-02, -8.6060e-03],
         [-1.0864e-02, -1.6724e-02,  1.1169e-02,  ..., -8.0566e-03,
           1.2512e-02,  1.7822e-02],
         [ 2.9755e-03, -8.3618e-03, -6.1646e-03,  ...,  1.2451e-02,
          -1.5869e-02, -9.5215e-03]],

        [[ 2.0020e-02,  1.5747e-02,  5.5695e-04,  ...,  1.7456e-02,
          -5.8899e-03,  2.6855e-03],
         [ 3.7384e-03, -6.5918e-03, -2.0752e-03,  ..., -1.7548e-03,
          -9.0942e-03, -4.8218e-03],
         [-7.9956e-03, -8.3542e-04,  9.4604e-04,  ..., -3.9062e-03,
           1.1719e-02,  8.9645e-05],
         ...,
         [-2.2125e-04, -7.3242e-03,  2.9907e-02,  ..., -3.4485e-03,
           5.5908e-02, -8.6060e-03],
         [-1.0864e-02, -1.6724e-02,  1.1169e-02,  ..., -8.0566e-03,
           1.2512e-02,  1.7822e-02],
         [ 2.9755e-03, -8.3618e-03, -6.1646e-03,  ...,  1.2451e-02,
          -1.5869e-02, -9.5215e-03]],

        [[ 2.0020e-02,  1.5747e-02,  5.5695e-04,  ...,  1.7456e-02,
          -5.8899e-03,  2.6855e-03],
         [ 3.7384e-03, -6.5918e-03, -2.0752e-03,  ..., -1.7548e-03,
          -9.0942e-03, -4.8218e-03],
         [-7.9956e-03, -8.3542e-04,  9.4604e-04,  ..., -3.9062e-03,
           1.1719e-02,  8.9645e-05],
         ...,
         [-2.2125e-04, -7.3242e-03,  2.9907e-02,  ..., -3.4485e-03,
           5.5908e-02, -8.6060e-03],
         [-1.0864e-02, -1.6724e-02,  1.1169e-02,  ..., -8.0566e-03,
           1.2512e-02,  1.7822e-02],
         [ 2.9755e-03, -8.3618e-03, -6.1646e-03,  ...,  1.2451e-02,
          -1.5869e-02, -9.5215e-03]]], device='cuda:0', dtype=torch.bfloat16), None, tensor([[ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80]], device='cuda:0', dtype=torch.int32), tensor([[ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80],
        [ 0, 80]], device='cuda:0', dtype=torch.int32), tensor([80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80,
        80, 80], device='cuda:0'), tensor([80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80,
        80, 80], device='cuda:0'), 0.0, 0.08838834764831845, False, True, False, None
wandb: Waiting for W&B process to finish... (failed 1). Press Control-C to abort syncing.
```